src/test/java/co/rsk/federate/FederationWatcherTest.java

```diff
@@ -22,354 +22,342 @@
     }
 
     @Test
-    void triggersActiveFederationChange_none_to_active() throws Exception {
-        EthereumListenerAdapter rskListener = setupAndGetRskListener(Optional.empty(), Optional.empty());
-        class EventsLogger
-        {
-            public int activeCalls = 0;
-            public int retiringCalls = 0;
-        }
-
-        when(federationProvider.getActiveFederationAddress()).thenReturn(federation1.getAddress());
-        when(federationProvider.getActiveFederation()).thenReturn(federation1);
+    void whenNoActiveFederation_shouldTriggerActiveFederationChange() throws Exception {
+        // Arrange
+        var rskListener = setupAndGetRskListener(Optional.empty(), Optional.empty());
+        var activeCalls = new AtomicInteger(0);
+        var retiringCalls = new AtomicInteger(0);
+
+        when(federationProvider.getActiveFederationAddress()).thenReturn(FIRST_FEDERATION.getAddress());
+        when(federationProvider.getActiveFederation()).thenReturn(FIRST_FEDERATION);
         when(federationProvider.getRetiringFederationAddress()).thenReturn(Optional.empty());
 
-        EventsLogger logger = new EventsLogger();
-
-        for (int i = 0; i < 2; i++) {
-            watcher.addListener(new FederationWatcher.Listener() {
-                @Override
-                public void onActiveFederationChange(Optional<Federation> oldFederation, Federation newFederation) {
-                    assertEquals(Optional.empty(), oldFederation);
-                    assertEquals(federation1, newFederation);
-                    logger.activeCalls++;
-                }
-
-                @Override
-                public void onRetiringFederationChange(Optional<Federation> oldFederation, Optional<Federation> newFederation) {
-                    logger.retiringCalls++;
-                }
-            });
-        }
-
+        federationWatcher.addListener(new FederationWatcher.Listener() {
+            @Override
+            public void onActiveFederationChange(Optional<Federation> oldFederation, Federation newFederation) {
+                assertEquals(Optional.empty(), oldFederation);
+                assertEquals(FIRST_FEDERATION, newFederation);
+                activeCalls.incrementAndGet();
+            }
+
+            @Override
+            public void onRetiringFederationChange(Optional<Federation> oldFederation, Optional<Federation> newFederation) {
+                retiringCalls.incrementAndGet();
+            }
+        });
+
+        // Act
         rskListener.onBestBlock(null, null);
-        assertEquals(2, logger.activeCalls);
-        assertEquals(0, logger.retiringCalls);
-        verify(federationProvider, times(1)).getActiveFederationAddress();
-        verify(federationProvider, times(1)).getRetiringFederationAddress();
-        verify(federationProvider, times(1)).getActiveFederation();
+
+        // Assert
+        assertEquals(1, activeCalls.get());
```

Comment:
wasn't asserting the active call were 2 before?

Reply:
Yes this was being done before. It was testing the same "code" twice since the listeners have the same implementation. So it leads to redundancy, not sure why it was designed like this as a whole :)